src/_P022_PCA9685.ino

```diff
@@ -52,38 +67,203 @@ boolean Plugin_022(byte function, struct EventStruct *event, String& string)
     case PLUGIN_WRITE:
       {
         String log = "";
-        String command = parseString(string, 1);
+        String line = String(string);
+        String command = "";
+        int dotPos = line.indexOf('.');
+        bool istanceCommand = false;
+        if(dotPos > -1)
+        {
+          LoadTaskSettings(event->TaskIndex);
+          String name = line.substring(0,dotPos);
+          name.replace("[","");
+          name.replace("]","");
+          if(name.equalsIgnoreCase(ExtraTaskSettings.TaskDeviceName) == true)
+          {
+            line = line.substring(dotPos + 1);
+            istanceCommand = true;
+          }
+          else
+          {
+             break;
+          }
+        }
+        command = parseString(line, 1);
 
-        if (command == F("pcapwm"))
+        if (command == F("pcapwm") || (istanceCommand && command == F("pwm")))
         {
-          if (!Plugin_022_init) Plugin_022_initialize();
           success = true;
-          Plugin_022_Write(event->Par1, event->Par2);
-          setPinState(PLUGIN_ID_022, event->Par1, PIN_MODE_PWM, event->Par2);
-          log = String(F("PCA  : GPIO ")) + String(event->Par1) + String(F(" Set PWM to ")) + String(event->Par2);
-          addLog(LOG_LEVEL_INFO, log);
-          SendStatus(event->Source, getPinStateJSON(SEARCH_PIN_STATE, PLUGIN_ID_022, event->Par1, log, 0));
+          log = String(F("PCA 0x")) + String(PCA9685_ADDRESS + port, HEX) + String(F(": GPIO ")) + String(event->Par1);
+          if(event->Par1>=0 && event->Par1 <= PCA9685_MAX_PINS)
+          {
+            if(event->Par2>=0 && event->Par1 <= PCA9685_MAX_PWM)
```

Comment:
Is this correct? One would expect `event->Par2` on both compares in the same if statement.

Reply:
yes, it's a mistake, thanks